src/app/core/admin/admin-dashboard/admin-dashboard.component.ts

```diff
@@ -0,0 +1,209 @@
+  imports: [
+    DynamicComponentDirective,
+    DragDropModule,
+    FaIconComponent,
+    FormsModule,
+    MatButtonModule,
+    MatCardModule,
+    MatTooltipModule,
+    MatIconModule,
+    ViewTitleComponent,
+  ],
+  templateUrl: "./admin-dashboard.component.html",
+  styleUrls: [
+    "./admin-dashboard.component.scss",
+    "../../dashboard/dashboard/dashboard.component.scss",
+    "../admin-entity/admin-entity-styles.scss",
+  ],
+})
+export class AdminDashboardComponent implements OnInit {
+  @Input() dashboardViewId: string;
+  @Input() isDisabled: boolean = false;
+
+  dashboardConfig: DashboardConfig;
+  private originalDashboardConfig: DashboardConfig;
+
+  private readonly configService = inject(ConfigService);
+  private readonly dialog = inject(MatDialog);
+  private readonly location = inject(Location);
+  private readonly snackBar = inject(MatSnackBar);
+
+  ngOnInit() {
+    this.loadDashboardViewConfig();
+  }
+
+  drop(event: CdkDragDrop<any[]>) {
+    if (event.previousContainer === event.container) {
+      moveItemInArray(
+        this.dashboardConfig.widgets,
+        event.previousIndex,
+        event.currentIndex,
+      );
+    } else {
+      // Handle drop from available widgets
+      const widgetType = event.previousContainer.data[event.previousIndex];
+      const newWidget: DynamicComponentConfig = {
+        component: widgetType.component,
+        config: { ...widgetType.defaultConfig },
+      };
+
+      this.dashboardConfig.widgets.splice(event.currentIndex, 0, newWidget);
+    }
+  }
+
+  async editWidget(widgetConfig: DynamicComponentConfig, idx: number) {
+    const settingsComponent = this.getSettingsComponentforWidget(widgetConfig);
+    if (!settingsComponent) return;
+
+    const updatedConfig = await this.openWidgetSettingsDialog(
+      widgetConfig,
+      settingsComponent,
+    );
+    if (updatedConfig) {
+      this.dashboardConfig.widgets[idx] = updatedConfig;
+      this.dashboardConfig.widgets = [...this.dashboardConfig.widgets];
+    }
+  }
+
+  getSettingsComponentforWidget(widgetConfig: DynamicComponentConfig): any {
+    switch (widgetConfig.component) {
+      case "ShortcutDashboard":
+        return "ShortcutDashboardSettings";
+
+      case "EntityCountDashboard":
+        return "EntityCountDashboardSettings";
+
+      case "ImportantNotesDashboard":
+        return "ImportantNotesDashboardSettings";
+
+      case "TodosDashboard":
+        return "TodosDashboardSettings";
+
+      case "NotesDashboard":
+        return "NotesDashboardSettings";
+
+      case "AttendanceWeekDashboard":
+        return "AttendanceWeekDashboardSettings";
+
+      case "ProgressDashboard":
+        return "ProgressDashboardSettings";
+
+      case "BirthdayDashboard":
+        return "BirthdayDashboardSettings";
+
+      default:
+        return null;
+    }
+  }
+
+  private prettifyWidgetTitle(componentName: string): string {
+    return componentName.replace(/([a-z])([A-Z])/g, "$1 $2");
+  }
```

Comment:
I think this can be handled in `AdminWidgetDialogComponent` if we don't need the title anywhere else here